src/npm.js

```diff
@@ -27,14 +27,15 @@ class Workspace {
   */
   async extract(url, target) {
     debug('extract', { url, target });
-    let path = fsPath.join(this.dir, 'node_modules.tar.gz');
+    let path = fsPath.join(target, 'node_modules.tar.gz');
 
-    let req = download({ extract: true }).
+    let req = download().
       get(url).
       dest(target).
-      use(downloadStatus());
+      use(downloadStatus())
 
     await denodeify(req.run).call(req);
+    await run('tar', ['zxf', path], { cwd: this.dir });
```

Comment:
Will do, using the 'extract' feature doesn't work very well and we also use 'tar' to compress in the first place. We'll have to find a better solution for that for adding Windows support so we can take care of it then.